src/actix/actix_telemetry.rs

```diff
@@ -19,6 +19,29 @@ pub struct ActixTelemetryTransform {
     telemetry_collector: Arc<Mutex<ActixTelemetryCollector>>,
 }
 
+// The uri will be `/collections/collection_name...`
+fn get_collection_from_uri(uri: &str) -> Option<String> {
+    let mut uri_segments = uri.split('/');
+    let seg0 = uri_segments.next();
+    if let Some(seg0) = seg0 {
+        if !seg0.is_empty() {
+            return None;
+        }
+    } else {
+        return None;
+    }
+    let seg1 = uri_segments.next();
+    if let Some(seg1) = seg1 {
+        if seg1 != "collections" {
+            return None;
+        }
+    } else {
+        return None;
+    }
+    let seg2 = uri_segments.next();
+    seg2.map(|seg2| seg2.to_owned())
+}
```

Comment:
Maybe we can use [`strip_prefix("/collections/")`](https://doc.rust-lang.org/std/primitive.str.html#method.strip_prefix) and [`split_once('/')`](https://doc.rust-lang.org/std/primitive.str.html#method.split_once) here to make it a bit more concise.